archive/extract_activity_test.go

```diff
@@ -0,0 +1,151 @@
+			params: archive.ExtractActivityParams{
+				SourcePath: filepath.Join("testdata", "transfer_subdir+file.zip"),
+			},
+			wantFs: tfs.Expected(t,
+				tfs.WithDir("subdir", tfs.WithMode(0o700)),
+				tfs.WithFile("small.txt",
+					`I am a small file.
+`,
+					tfs.WithMode(0o600),
+				),
+			),
+		},
+		{
+			name: "Extracts a zip archive with explicit file modes",
+			cfg:  archive.Config{DirMode: 0o750, FileMode: 0o640},
+			params: archive.ExtractActivityParams{
+				SourcePath: filepath.Join("testdata", "transfer_subdir+file.zip"),
+			},
+			wantFs: tfs.Expected(t,
+				tfs.WithDir("subdir", tfs.WithMode(0o750)),
+				tfs.WithFile("small.txt",
+					`I am a small file.
+`,
+					tfs.WithMode(0o640),
+				),
+			),
+		},
+		{
+			name: "Extracts a 7z archive",
+			params: archive.ExtractActivityParams{
+				SourcePath: filepath.Join("testdata", "transfer.7z"),
+			},
+			wantFs: tfs.Expected(t,
+				tfs.WithFile("small.txt",
+					`I am a small file.
+`,
+					tfs.WithMode(0o600)),
+			),
+		},
+		{
+			name: "Errors when SourcePath is a dir",
+			params: archive.ExtractActivityParams{
+				SourcePath: "testdata",
+			},
+			wantErr: fmt.Sprintf(
+				"activity error (type: extract-archive-activity, scheduledEventID: 0, startedEventID: 0, identity: ): %s",
+				archive.ErrNotAFile,
+			),
+		},
+		{
+			name: "Errors when SourcePath is a non-archive file",
+			params: archive.ExtractActivityParams{
+				SourcePath: filepath.Join("testdata", "small.txt"),
+			},
+			wantErr: fmt.Sprintf(
+				"activity error (type: extract-archive-activity, scheduledEventID: 0, startedEventID: 0, identity: ): %s",
+				archive.ErrInvalidArchive,
+			),
+		},
+		{
+			name: "Errors on corrupt archive",
+			params: archive.ExtractActivityParams{
+				SourcePath: filepath.Join("testdata", "corrupt.zip"),
+			},
+			wantErr: "activity error (type: extract-archive-activity, scheduledEventID: 0, startedEventID: 0, identity: ): extract archive: extract: zip: not a valid zip file",
+		},
+		{
+			name: "Errors when source path doesn't exist",
+			params: archive.ExtractActivityParams{
+				SourcePath: filepath.Join("testdata", "missing.zip"),
+			},
+			wantErr: "activity error (type: extract-archive-activity, scheduledEventID: 0, startedEventID: 0, identity: ): extract archive: stat testdata/missing.zip: no such file or directory",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			ts := &temporalsdk_testsuite.WorkflowTestSuite{}
+			env := ts.NewTestActivityEnvironment()
+			env.RegisterActivityWithOptions(
+				archive.NewExtractActivity(tt.cfg).Execute,
+				temporalsdk_activity.RegisterOptions{
+					Name: archive.ExtractActivityName,
+				},
+			)
+
+			enc, err := env.ExecuteActivity(archive.ExtractActivityName, &tt.params)
+			if tt.wantErr != "" {
+				assert.Error(t, err, tt.wantErr)
+				return
+			}
+			assert.NilError(t, err)
+
+			var result archive.ExtractActivityResult
+			_ = enc.Get(&result)
```

Comment:
I'd be nice to check the activity result too.

Reply:
We are. If `result.DestPath` is empty, or incorrect then the following assert will fail:
```go
assert.Assert(t, tfs.Equal(result.DestPath, tt.wantFs))
```

I originally had a test that `result.DestPath` is not an empty string, but I cut it because it doesn't add any extra assurance of correctness, imho.